src/c++11/propagation.h

```diff
@@ -0,0 +1,93 @@
+// -*- Mode: C++ -*-
+#ifndef __LIGHTSTEP_PROPAGATION_H__
+#define __LIGHTSTEP_PROPAGATION_H__
+
+#include <functional>
+#include <string>
+
+namespace lightstep {
+  
+enum BuiltinFormat {
+  // Binary encodes the SpanContext for propagation as opaque binary data.
+  Binary = 1,  // NOT IMPLEMENTED
+
+  // TextMap encodes the SpanContext as key:value pairs.
+  TextMap = 2,  // NOT IMPLEMENTED
+
+  // HttpHeaders encodes the SpanContext as HTTP headers.
+  HttpHeaders = 3,
+};
+
+// Base class for implementation-provided or builtin carrier formats.
+class CarrierFormat {
+public:
+};
+
+// Builtin carrier format values.
+class BuiltinCarrierFormat : public CarrierFormat {
```

Comment:
We could replace the enum with a string, if that's what you're asking. The enum is handy if you want a switch statement, and the class is handy if you want a dynamic_cast. A string is self-describing. I don't know (or think it matters much).